presidio-image-redactor/tests/integration/test_dicom_image_pii_verify_engine_integration.py

```diff
@@ -0,0 +1,132 @@
+"""Integration test for dicom_image_pii_verify_engine
+
+Note we are not checking exact pixel data for the returned image
+because that is covered by testing of the "verify" function in
+the original parent ImagePiiVerifyEngine class.
+"""
+import PIL
+import pydicom
+import json
+import os
+
+from presidio_image_redactor.dicom_image_pii_verify_engine import (
+    DicomImagePiiVerifyEngine,
+)
+import pytest
+
+SCRIPT_DIR = os.path.dirname(__file__)
+PADDING_WIDTH = 25
+
+
+@pytest.fixture(scope="module")
+def mock_engine():
+    """Instance of the DicomImagePiiVerifyEngine"""
+    # Arrange
+
+    # Act
+    dicom_image_pii_verify_engine = DicomImagePiiVerifyEngine()
+
+    return dicom_image_pii_verify_engine
+
+
+@pytest.fixture(scope="module")
+def mock_instance():
+    """DICOM instance to use in testing"""
+    # Assign
+    filepath = f"{SCRIPT_DIR}/resources/0_ORIGINAL.dcm"
+
+    # Act
+    instance = pydicom.dcmread(filepath)
+
+    return instance
+
+
+@pytest.fixture(scope="module")
+def mock_results():
+    """Loaded json results file"""
+    with open(f"{SCRIPT_DIR}/resources/dicom_pii_verify_integration.json") as json_file:
+        results_json = json.load(json_file)
+
+    return results_json
+
+
+def test_verify_correctly(
+    mock_engine: DicomImagePiiVerifyEngine,
+    mock_instance: pydicom.dataset.FileDataset,
+    mock_results: dict,
+):
+    """Test the verify_dicom_instance function.
+
+    Args:
+        mock_engine (DicomImagePiiVerifyEngine): Instantiated engine.
+        mock_instance (pydicom.dataset.FileDataset): Loaded DICOM.
+        mock_results (dict): Dictionary with loaded results.
+    """
+    # Assign
+    expected_ocr_results = mock_results["ocr_results_formatted"]
+    expected_analyzer_results = mock_results["analyzer_results"]
+    expected_ocr_results_labels = []
+    for i in expected_ocr_results:
```

Comment:
for item in expected_ocr_results:

Reply:
or even better `expected_ocr_results_labels = [item["label"] for item, in expected_ocr_results]`

Reply:
Doing so just returns the key for item. For example
```python
tmp = {"1": {"2": [1,2,3], "3": [4,5,6]}}
for item in tmp:
    print(item)
```
Prints out 1 and not the contents of 1.

Reply:
Disregard. I have switched over to using lists instead of a dictionary so this now works.